fields.f90

```diff
@@ -6525,7 +6527,7 @@ subroutine Lmat_generation1d
          !
          kindex = 0 ; kindex(imode) = 1
          !
-         coordtransform(:,imode) = FLvect_finitediffs(job_is,trove%Ncoords,kindex,q_eq,step,irho)
+         coordtransform(:,imode) = FLvect_finitediffs(job_is,trove%Nmodes,kindex,q_eq,step,irho)
```

Comment:
This seems like a big thing to have been missed for so long! Do the benchmark problems have `Nmodes == Ncoords`?

Reply:
The changed argument actually sets the size of the output of the function `FLvect_finitediffs`, which is then assigned to `coordtransform`. The first dim of `coordtransform` is of length `trove%Nmodes` which is inconsistent with the value passed. In this situation, intel fortran was broadcasting only what fit into the output.

Reply:
hi Jamie, 
Please be careful with this parameter.  I can see that there is a conflict with the definition of the  array coordtransform. However trove%Ncoords was correct here. I am pretty sure using trove%Nmodes will break cases where trove%Nmodes/=trove%Ncoords such as CH4. 
I will try to understand the source of this bug. Most likely it is the definition of coordtransform.

Reply:
Never mind, Jamie. I think you are right and trove%Nmodes should give the correct size for these vectors.